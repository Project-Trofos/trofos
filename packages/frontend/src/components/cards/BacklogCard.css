li.backlog-card-container {
    background: #fff;
    border: 1px solid #eee;
    padding: 12px;
}

.backlog-card-container div {
    margin-right: 8px;
}

.backlog-card-summary {
    display: flex;
    flex: 1 1 0%;
}