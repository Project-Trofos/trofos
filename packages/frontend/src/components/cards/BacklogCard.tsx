import React from 'react';
import type { Backlog } from '../../api/backlog';
import './BacklogCard.css';

function BacklogCard(props: BacklogCardProps): JSX.Element {
  const { backlog } = props;
  return (
    <>
      <div>{backlog.id}</div>
      <div className="backlog-card-summary">{backlog.summary}</div>
      <div>{backlog.type}</div>
      <div>{backlog.priority}</div>
      <div>{backlog.assignee_id}</div>
      <div>{backlog.points}</div>
    </>
  );
}

type BacklogCardProps = {
  backlog: Backlog;
};

export default BacklogCard;
