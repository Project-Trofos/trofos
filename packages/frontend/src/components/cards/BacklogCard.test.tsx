import React from 'react';
import { render, screen } from '@testing-library/react';
import BacklogCard from './BacklogCard';
import type { Backlog } from '../../api/backlog';

describe('BacklogCard test', () => {
  const mockBacklog: Backlog = {
    id: 111,
    summary: 'A Test Summary',
    type: 'story',
    priority: 'very_high',
    sprint_id: 123,
    reporter_id: 1,
    assignee_id: 2,
    points: 3,
    description: 'A test description here',
    project_id: 123,
  };
  const { baseElement } = render(<BacklogCard backlog={mockBacklog} />);

  it('renders backlog card with correct details', () => {
    expect(screen.getByText('111')).toBeInTheDocument();
    expect(screen.getByText('A Test Summary')).toBeInTheDocument();
    expect(screen.getByText('story')).toBeInTheDocument();
    expect(screen.getByText('very_high')).toBeInTheDocument();
    expect(screen.getByText('User2')).toBeInTheDocument();
    expect(screen.getByText('3')).toBeInTheDocument();

    // Compare with snapshot to ensure structure remains the same
    expect(baseElement).toMatchSnapshot();
  });
});
