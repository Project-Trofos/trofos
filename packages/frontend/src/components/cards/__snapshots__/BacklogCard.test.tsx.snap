// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BacklogCard test renders backlog card with correct details 1`] = `
<body>
  <div>
    <div>
      111
    </div>
    <div
      class="backlog-card-summary"
    >
      A Test Summary
    </div>
    <div>
      story
    </div>
    <div>
      very_high
    </div>
    <div>
      222
    </div>
    <div>
      3
    </div>
  </div>
</body>
`;
